person.rb

```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(name: 'Unknown', age: 'nil', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age || 0
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age? || parent_permission
+  end
```

Comment:
- [ ] Great logic was used to create this method. However, if you call the method, it will give you this error:

![image](https://github.com/vanina123/School-libary/assets/77106962/ea99dde6-4c18-4949-acfe-73ca798579d5)

That's because `parent_permission` is a **_local variable_** while `@parent_permission` is an **_instance variable_**. The difference is that the **_local variable_** can only be seen within a method if it is not declared in the general scope **_(method/local scope)_** while the **_instance variable_** can be accessed from anywhere within the class **_(class scope)_**. Kindly implement this change. Kindly use the instance variable `@parent_permission` in this method. 👍 💻